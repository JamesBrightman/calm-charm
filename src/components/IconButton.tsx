import { LoadingButton } from '@mui/lab';
import { ButtonProps } from '@mui/material';
import React, { FC } from 'react';

export interface IconButtonProps extends ButtonProps {
  icon: JSX.Element;
  loading?: boolean;
  buttonText: string;
  handleClick?: () => void;
}

export const IconButton: FC<IconButtonProps> = ({
  icon,
  loading,
  buttonText,
  handleClick,
  className,
  ...props
}) => {
  return (
    <LoadingButton
      className={
        'grid grid-flow-col grid-cols-[50px_1fr] items-center justify-items-center rounded-xl border-2 px-1' +
        ' ' +
        className +
        ' ' +
        `${loading ? 'bg-opacity-50 bg-gray-400' : 'bg-opacity-100'}`
      }
      {...props}
      onClick={handleClick}
      loading={loading}
      loadingPosition="end"
    >
      {icon}
      <p className="pr-[50px]">{buttonText}</p>
    </LoadingButton>
  );
};
