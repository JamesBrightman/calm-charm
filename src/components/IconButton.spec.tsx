import React from 'react';
import { render, screen } from '@testing-library/react';
import { IconButton } from './IconButton';

describe('Icon Button Component', () => {
  it('Icon button renders', () => {
    const { baseElement } = render(
      <IconButton icon={<div>Test Icon</div>} buttonText={'Foo'} />
    );

    expect(baseElement).toBeTruthy();
  });

  it('Icon button matches snapshot', () => {
    const { baseElement } = render(
      <IconButton icon={<div>Test Icon</div>} buttonText={'Foo'} />
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('Icon button has custom text', () => {
    render(<IconButton icon={<div>Test Icon</div>} buttonText={'Bar Foo'} />);

    expect(screen.getByText('Bar Foo')).toBeInTheDocument();
  });

  it('Icon button renders custom class', () => {
    render(
      <IconButton
        className="bg-green-500 text-sm"
        icon={<div>Test Icon</div>}
        buttonText={'Foo'}
        data-testId="iconId"
      />
    );

    expect(screen.getByTestId('iconId')).toHaveClass('bg-green-500 text-sm');
  });
});
