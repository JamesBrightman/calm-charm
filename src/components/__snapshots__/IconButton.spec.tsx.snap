// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Button Component Icon button matches snapshot 1`] = `
<body>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiLoadingButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium grid grid-flow-col grid-cols-[50px_1fr] items-center justify-items-center rounded-xl border-2 px-1 undefined bg-opacity-100 css-1tw7cpj-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
      id=":r1:"
      tabindex="0"
      type="button"
    >
      <div>
        Test Icon
      </div>
      <p
        class="pr-[50px]"
      >
        Foo
      </p>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</body>
`;
